virtual-scroller-element.js

```diff
@@ -85,52 +92,70 @@ export class VirtualScrollerElement extends HTMLElement {
     this.setAttribute('layout', layout);
   }
 
-  get totalItems() {
-    return +this.getAttribute('totalitems');
+  get itemSource() {
+    return this[_itemSource];
   }
-  set totalItems(v) {
-    this.setAttribute('totalitems', +v);
+  set itemSource(itemSource) {
+    // No Change.
+    if (this[_rawItemSource] === itemSource) {
+      return;
+    }
+    this[_rawItemSource] = itemSource;
+    this[_itemSource] = Array.isArray(itemSource) ?
+        ItemSource.fromArray(itemSource) :
+        itemSource;
+    this[_render]();
   }
 
   get createElement() {
     return this[_createElement];
   }
   set createElement(fn) {
+    // No change.
+    if (fn === this[_createElement]) {
```

Comment:
In general I'm not sure of the value of these kind of guards.

Reply:
Avoid rendering. In particular `createElement` changes cause the internal key/Element map to be invalidated.

Reply:
Sure. So don't set createElement if you don't want things to be re-rendered, is my argument. (But, if there are popular frameworks which make that hard, then we need to think about it. Let's discuss in the above thread about img.src etc.)

Reply:
sg, let me remove these